### ISSUE TITLE
OpenEBench summary workflow: improve bedtools error handling during metrics calculation

### ISSUE THREAD

**Issue by AsierGonzalez** [iRNA-COSI/APAeval#146]
We have found that the metrics calculation step fails badly if the input file does not match the reference file at all. It produces a this not very useful error:

```
 Traceback (most recent call last):
    File "/app/compute_metrics.py", line 105, in <module>
      main(args)
    File "/app/compute_metrics.py", line 32, in main
      compute_metrics(input_participant, gold_standards_dir, challenge_types, participant, community, out_path)
    File "/app/compute_metrics.py", line 61, in compute_metrics
      match_with_gt_run = Leo_MatchPAsites_GT.match_wtih_gt(input_participant,metrics_data,window)
    File "/app/Leo_MatchPAsites_GT/Leo_MatchPAsites_GT.py", line 52, in match_wtih_gt
      out = bedtools_window(f_PD, f_GT, window)
    File "/app/Leo_MatchPAsites_GT/Leo_MatchPAsites_GT.py", line 32, in bedtools_window
      assert out.returncode==0, "bedtools window run failed, check input files"
  AssertionError: bedtools window run failed, check input files
```
Better handling of non-overlapping BEDs. e.g. from different organisms, genome builds, etc., may be needed here.

**Comment by yuukiiwa** [comment]
Hi @AsierGonzalez, I was testing the samples with their respective ground truth files (all samples except M_R6) and didn't encounter this error unless the reference file path couldn't be found. I will post the local runs of the summary workflow in the OEB channel on Slack. Thanks!

**Comment by AsierGonzalez** [comment]
What if you run a sample with a reference that isn't its own? I believe that this is what happened to me when I was testing a file without realising that the reference file was hardcoded.

**Comment by yuukiiwa** [comment]
Hi @AsierGonzalez, I have tried mixing and matching the `bed` input files and the ground-truth files so that none of them are matching to their respective ground truth, but I couldn't see the error:
GC R2 (QAPA) vs CD5B R4 ground truth:
![Screenshot 2021-06-28 at 6 00 44 PM](https://user-images.githubusercontent.com/41866052/123619566-e1ece180-d83b-11eb-9ca2-a41c26c8f0fe.png)
M R2 (QAPA) vs NB group Multiple SRR ground truth:
![Screenshot 2021-06-28 at 6 05 43 PM](https://user-images.githubusercontent.com/41866052/123619672-fcbf5600-d83b-11eb-9d4b-7d45611b09d1.png)
CD5B R3 (LABRAT) vs GC group Multiple SRR ground truth:
![Screenshot 2021-06-28 at 5 58 11 PM](https://user-images.githubusercontent.com/41866052/123619756-16609d80-d83c-11eb-9a33-7ea6f5d0021b.png)
NB R5 (QAPA) vsM R2 ground truth:
![Screenshot 2021-06-28 at 6 06 06 PM](https://user-images.githubusercontent.com/41866052/123619934-43ad4b80-d83c-11eb-96c5-1b86cb0068ba.png)
[Here](https://github.com/yuukiiwa/APAeval-summary-workflow/blob/d1fcea6e360fc09b4fe77dfb7284c71fb7c7ddac/nextflow.config#L52) is the line where I indicate the tag of the ground truth file (without `.bed` because the metrics calculation script will add `.bed` to it), and I have to change the path to the input file and the tag of the ground truth file in `nextflow.config` each time running different samples:
```
// submitted file
  input = "$baseDir/APAeval_Q2_test_data/Mayr_NB_R5.qapaFixed.bed"

  // reference file dir to validate input data
  public_ref_dir = "$baseDir/APAeval_Q2_test_data/public_ref"

  // name of the tool used for the predicitions
  participant_id = "test"

  // directory where the 'gold standards' are found
  goldstandard_dir = "$baseDir/APAeval_Q2_test_data/metrics_ref_datasets"

  // challenge types for which the benchmark has to be performed
  challenges_ids  = "Mayr_M_R2.SRR6795691.3seq.hg38"
```
I accidentially recreated the error you saw by including `.bed` in the challenge_id:
```
  challenges_ids  = "Mayr_M_R2.SRR6795691.3seq.hg38.bed"
```
Thanks!

**Comment by ninsch3000** [comment]
Will be resolved in #182
